_posts/moving-rails-apps-off-webpacker.md

```diff
@@ -0,0 +1,92 @@
+
+On the Research Data and Scholarship Services (RDSS) team we utilized Vite after DLS and DACS each converted one of thier repositories. [pdc_describe](https://github.com/pulibrary/pdc_describe/pull/397/files) was converted to utilize vite and the transition went smoothly.
+
+#### Vite tips / lessons learned
+
+- Learn to pronounce it!: https://vitejs.dev/guide/#overview
+- Vite does not allow for (or need) testing assets to be served from the [development server](https://github.com/pulibrary/pdc_describe/pull/403).
+- If you import any files that have a file extension other than `.js`, you will need to include
+the file extension when importing it.  We had a few files with an `.es6` file extension, so we
+had to change this:
+
+      import BookCoverManager from '../orangelight/book_covers'
+
+    to this:
+
+      import BookCoverManager from '../orangelight/book_covers.es6'
+
+#### But what's it like to work with?
+
+Migration is only the first step, right? On DLS, we haven't had a work cycle yet
+with the app we moved to vite so we can't speak much to the developer
+experience. It is straightforward to run `bin/vite dev` and have autoreloading
+if you don't want to refresh your browser to see changes.
+
+On DACS, we have made some small changes to Orangelight's javascript since moving to vite,
+and haven't noticed much difference from our previous experience with webpacker.
+
+### Importmaps
+
+Importmaps is supposed to allow you to manage your javascript dependencies
+without having node installed. It's the simplest of the new options so on DLS we
+tried it with our simplest rails app. Like all our other apps, this one uses our
+vue.js-based design system, [lux](https://github.com/pulibrary/lux). We got
+pretty far with importmaps (and dartsass-rails), but we weren't able to get this
+integration with our design system. My speculation is that lux would need to
+provide a node package that bundles the full source for all its dependencies? This is partially based on a [DHH
+post](https://discuss.rubyonrails.org/t/rails-es6-based-replacement-for-webpacker/78656/7)
+where he says, "I’d love to see a method that downloads the pinned dependencies
+at some point, but it’s not trivial for packages that have nested dependencies,
+which haven’t been bundled into a single dist file."
+
+### Jsbundling-rails with esbuild
+
+DLS tested this migration on Figgy, our largest app. The initial upgrades went smoothly and there was lots of primary and secondary documentation online. The configuration ended up feeling awkward and we decided to use Vite instead.
+
+### Shakapacker
+
+We tried shakapacker in [pdc_discovery](https://github.com/pulibrary/pdc_discovery/pull/307).  It went relatively smoothly, but we never decided to put it into production.
+
+### General tips / lessons learned
+
+- When loading both local javascript and Vue.js components, you have to be careful
+  about how you order them. Vue.js takes over the eventing system so once it's
```

Comment:
Does this concern apply to all the replacement options? If so, I'd add an intro comment like:
```suggestion
- Whichever webpacker replacement you choose, you have to be careful about how you order local javascript
and Vue.js components when loading both. Vue.js takes over the eventing system so once it's
```